### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by rafayet-monon**
https://github.com/nimblehq/elixir-templates/issues/118

## What happened

Added prettier to the template.
 
## Insight

- Add `prettier": "2.2.1` and `prettier-plugin-eex": "^0.5.0`
- Add .prettierrc.js and and `.prettierignore` to the template
- Add test codes
 
## Proof Of Work

- CI should pass.
- After applying the template and running `npm i` in assets the plugin is working nicely.
image

### CODE REVIEW

**Review by andyduong1920 — APPROVED** [review]
LGTM 👏

File: lib/nimble_template/addons/variants/phoenix/web/prettier.ex [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  defp edit_mix(%Project{} = project) do
+    Generator.inject_content(
+      "mix.exs",
+      """
+        defp aliases do
+          [
+      """,
+      """
+            prettier: "cmd ./assets/node_modules/.bin/prettier --write . --color",
```
Comment (rosle) [inline lib/nimble_template/addons/variants/phoenix/web/prettier.ex:45]: Is it possible to also add this to a list of codebase checks or to Github workflow? 🤔 💭

File: .tool-versions [hunk 2]
```diff
@@ -1,2 +1,2 @@
-erlang 23.3
```
Comment (olivierobert) [inline .tool-versions:1]: Why this change?

File: priv/templates/nimble_template/.prettierrc.js.eex [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (olivierobert) [inline priv/templates/nimble_template/.prettierrc.js.eex:1]: Since it's an Elixir project, can we use the file format `.prettierrc`, so dropping `.js`. 

It's supported by Prettier:

image

Instead of using `module.exports` then it could be a JSON or YAML 💡

**Comment by andyduong1920** [comment]
hi @rafayet-monon could you have a look on this PR whenever you have time 🙏

**Comment by andyduong1920** [comment]
Please rebase this PR @rafayet-monon  the test was fixed on #128 👍

**Review by andyduong1920 — APPROVED** [review]

**Review by rosle — APPROVED** [review]

**Review by junan — APPROVED** [review]
